airflow/www/static/js/dags.js

```diff
@@ -37,6 +37,20 @@ const lastDagRunsUrl = getMetaValue('last_dag_runs_url');
 const dagStatsUrl = getMetaValue('dag_stats_url');
 const taskStatsUrl = getMetaValue('task_stats_url');
 
+const FILTER_TAGS = 'filter_tags';
+
+const filterTags = localStorage.getItem(FILTER_TAGS);
+const initialQuery = new URLSearchParams(window.location.search);
+
+// Update the url if there is no tags query but localstorage has tags saved
+if (!initialQuery.has('tags') && filterTags) {
+  const tags = JSON.parse(filterTags);
+  tags.forEach((tag) => {
+    initialQuery.append('tags', tag);
+  });
+  window.location = `${DAGS_INDEX}?${initialQuery.toString()}`;
```

Comment:
I'm confused - This still involves a round trip

Reply:
Unfortunately, this one needed to stay. but reseting to no tags no longer requires 2 redirects.